pkg/serverless/trace/inferredspan/span_enrichment.go

```diff
@@ -11,23 +11,71 @@ import (
 	"strings"
 	"time"
 
-	"github.com/aws/aws-lambda-go/events"
-
+	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/aws/aws-lambda-go/events"
 )
 
+// Define and initialize serviceMapping as a global variable.
+var serviceMapping map[string]string
+
+func CreateServiceMapping(val string) map[string]string {
+	newServiceMapping := make(map[string]string)
+
+	for _, entry := range strings.Split(val, ",") {
+		parts := strings.Split(entry, ":")
+		if len(parts) == 2 {
+			key := strings.TrimSpace(parts[0])
+			value := strings.TrimSpace(parts[1])
+			if key != "" && value != "" && key != value {
+				newServiceMapping[key] = value
+			}
+		}
+	}
+	return newServiceMapping
+}
+
+func init() {
+	serviceMappingStr := config.Datadog.GetString("serverless.service_mapping")
+	serviceMapping = CreateServiceMapping(serviceMappingStr)
+}
+
+// SetServiceMapping sets the serviceMapping global variable, primarily for tests.
+func SetServiceMapping(newServiceMapping map[string]string) {
+	serviceMapping = newServiceMapping
+}
+
+// This function gets a snapshot of the current service mapping without modifying it.
+func GetServiceMapping() map[string]string {
+	return serviceMapping
+}
```

Comment:
Can we move this to the tests instead of having it here?

Reply:
I know I moved them in the python implementation but I think its best to have them here, it provides a way to change and access the state of serviceMapping within the tests. Otherwise if I moved them I would have to export serviceMapping (i.e. make it capitalized) and that could lead to other issues down the road of that variable being altered in other ways we didn't intend to. LMKWYK

Reply:
hmm well I guess someone could use SetServiceMapping and GetServiceMapping anyways to change serviceMapping since they are exported functions but at least this way we know they will interact with it through these defined ways